### PR TITLE
add state syncing registration

### PR DESCRIPTION
**Pull request by chaosma**
Add node registration for state syncing. If a node is out of sync, it will register itself to a couple of peers and waiting for new blocks to broadcast.

### CODE REVIEW

File: test/bc_config.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{
```
Comment (LeoHChen) [inline test/bc_config.json]: please remove bc_config.json

File: node/node.go [hunk 2]
```diff
@@ -119,9 +120,10 @@ type Node struct {
 	clientServer *clientService.Server
 
 	// Syncing component.
-	downloaderServer *downloader.Server
-	stateSync        *syncing.StateSync
-	syncingState     uint32
+	downloaderServer       *downloader.Server
+	stateSync              *syncing.StateSync
+	syncingState           uint32
+	peerRegistrationRecord map[string]int64 // record registration time (unixtime) of peers begin in syncing
```
Comment (LeoHChen) [inline node/node.go]: You may have map concurrent access issue. please consider using sync.Map data structure.

File: test/bc_config.json (hunk 1 above)
Comment (chaosma) [inline test/bc_config.json]: I didn't create this file

Comment (LeoHChen) [reply test/bc_config.json]: yes, it's a beacon chain by-product.  don't check it in.  we should actually add it to .gitingore.

File: node/node.go (hunk 2 above)
Comment (chaosma) [reply node/node.go]: I will add mutex for map success

**Pull request closed without merging**